### PR TITLE
Added functionality for c++

### LINKED ISSUE #5: CPP Executor
Current the [executor for cpp](https://github.com/benwoo1110/testcase-maker/blob/main/testcase_maker/executors/cpp.py) scripts is not done yet. The main task is for it to run cli commands to compile and execute a cpp code file with stdin. You can see the [java](https://github.com/benwoo1110/testcase-maker/blob/main/testcase_maker/executors/java.py) and [python](https://github.com/benwoo1110/testcase-maker/blob/main/testcase_maker/executors/python.py) executors for example.

### PR DESCRIPTION
**Pull request by daksh10110**
I have done the necessary stuff for it, but I am unable to test it, because the code written in readme and the files here aren't the same, the code over there is camel cased and code in the repo is under snake casing.

I am unable to figure out how to test it, so please test it and lemme know if something's wrong. #5

### CODE REVIEW

**Comment by benwoo1110** [comment]
Ok I will test. But I am not sure what you mean by "code written in readme and the files here aren't the same".

File: testcase_maker/executors/cpp.py [hunk 1]
```diff
@@ -1,23 +1,26 @@
...
+    Run your answer script in C++!
     """
 
     @property
     def file_extension(self) -> str:
         return "cpp"
 
     def compile(self, tempdir: Union["Path", str], source_filename: Union["Path", str]) -> Union["Path", str]:
-        pass
+        args = ["g++", "-o", str(source_filename.stem), str(source_filename)]
```
Comment (benwoo1110) [inline testcase_maker/executors/cpp.py]:
```suggestion
        args = ["g++", "-o", str(source_filename.stem), str(source_filename.absolute())]
```

Tested, need to use absolute path here.

File: testcase_maker/executors/cpp.py [hunk 2]
```diff
@@ -1,23 +1,26 @@
...
+        run_command(args, cwd=tempdir)
+        return tempdir.joinpath(f"{source_filename.stem}.exe")
 
     def execute(self, exec_filename: Union["Path", str], stdin: str) -> bytes:
-        pass
+        args = [str(exec_filename.stem) + ".exe"]
+        return run_command(args, stdin, exec_filename.parent)
```
Comment (benwoo1110) [inline testcase_maker/executors/cpp.py]: Prefer to have a newline at end of file.

File: testcase_maker/executors/cpp.py [hunk 3]
```diff
@@ -1,23 +1,26 @@
...
 
     def compile(self, tempdir: Union["Path", str], source_filename: Union["Path", str]) -> Union["Path", str]:
-        pass
+        args = ["g++", "-o", str(source_filename.stem), str(source_filename)]
+        run_command(args, cwd=tempdir)
+        return tempdir.joinpath(f"{source_filename.stem}.exe")
 
     def execute(self, exec_filename: Union["Path", str], stdin: str) -> bytes:
-        pass
+        args = [str(exec_filename.stem) + ".exe"]
```
Comment (benwoo1110) [inline testcase_maker/executors/cpp.py]:
```suggestion
        args = [str(exec_filename)]
```

**Comment by daksh10110** [comment]
I have fixed those issues, ~~by names aren't same I mean that,~~

~~on readme it's written~~
```
from testcase_maker.generator import TestcaseGenerator
from testcase_maker.values import ValueGroup, NamedValue, RandomInt, LoopValue, ValueRef
```

~~so basically I had to import `TestcaseGenerator` and `ValueGroup, NamedValue, RandomInt, LoopValue, ValueRef` from generator values respectively.~~

~~So TestcaseGenerator is over there as it should be but under values we have  `ValueGroup, NamedValue, RandomInt, LoopValue, ValueRef`  under camel casing which is `value_group, named_value, random_int, loop_value and value_ref`~~

~~I think that was the reason for my inability to test.~~

My bad, I didn't read that code properly before

**Review by benwoo1110 — APPROVED** [review]
👍

**Pull request merged**